osu.Framework/Graphics/UserInterface/DirectorySelectorDirectory.cs

```diff
@@ -27,7 +27,8 @@ protected DirectorySelectorDirectory(DirectoryInfo directory, string displayName
 
             try
             {
-                if (directory?.Attributes.HasFlagFast(FileAttributes.Hidden) == true)
+                if (directory?.Attributes.HasFlagFast(FileAttributes.System) == true) { }
+                else if (directory?.Attributes.HasFlagFast(FileAttributes.Hidden) == true)
```

Comment:
This should read better:

```suggestion
                bool isHidden = directory?.Attributes.HasFlagFast(FileAttributes.Hidden) == true;

                // System drives show up as `System | Hidden | Directory` but shouldn't be shown in a hidden state.
                bool isSystem = directory?.Attributes.HasFlagFast(FileAttributes.System) == true;

                if (isHidden && !isSystem)
```

Reply:
I dunno about using `IsSystem` here at all. From a semantic point of view there is no reason why a hidden system file should generally become unhidden.

I'd probably use `directoryInfo.Parent == null` or similar as it feels more targeted for catching the "this is actually a drive" case.

Reply:
Can agree with this. Seems less prone to failing.